feat: Cloud Gateway terminology updates (#7093)
Kong/docs.konghq.com: app/konnect/gateway-manager/index.md

A data plane node is a single {{site.base_gateway}} instance. 
Data plane nodes service traffic for the control plane. 

> Kong only hosts fully-managed Dedicated Cloud Gateway data plane nodes.
> You must deploy your own nodes if you want to self-manage them, either on your own systems or in 
> an external cloud provider.

Reviewer: ```suggestion
You can have deploy your data plane nodes in one of the following ways:
* Self-managed nodes, hosted either on your own systems or in 
an external cloud provider
* Fully-managed dedicated cloud gateway nodes
```
Rephrasing to make this sound more positive. The previous phrasing implied a lack of something.
